### PR TITLE
Hotkeys in SQL Lab

### PR DESCRIPTION
**Pull request by mistercrunch**
screen shot 2018-03-23 at 3 19 00 pm

### CODE REVIEW

**Comment by mistercrunch** [comment]
removed the [WiP] tag, this is ready for review

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx [hunk 1]
```diff
@@ -226,6 +229,32 @@ class SqlEditor extends React.PureComponent {
   render() {
     const height = this.sqlEditorHeight();
     const defaultNorthHeight = this.props.queryEditor.height || 200;
+    const SQLLAB_HOTKEYS = [
```
Comment (hughhhh) [inline superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: nit: Can we put this at the top of the file or even in config file for adding new hotkeys more easily. I'm like the idea of a having this separate and we import it

File: superset/assets/javascripts/SqlLab/actions.js [hunk 2]
```diff
@@ -196,7 +196,10 @@ export function setDatabases(databases) {
 }
 
 export function addQueryEditor(queryEditor) {
-  const newQe = Object.assign({}, queryEditor, { id: shortid.generate() });
+  const newQe = {
```
Comment (hughhhh) [inline superset/assets/javascripts/SqlLab/actions.js]: nit: neQe >> newQuery

File: superset/assets/javascripts/components/Hotkeys.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    this.props.hotkeys.forEach((keyConfig) => {
+      Mousetrap.bind([keyConfig.key], keyConfig.func);
+    });
+  }
+  renderPopover() {
+    return (
+      <Popover id="popover-hotkeys" title={this.props.header} style={{ width: '300px' }}>
+        <Table
+          className="table table-condensed"
+          data={this.props.hotkeys.map(keyConfig => ({
```
Comment (hughhhh) [inline superset/assets/javascripts/components/Hotkeys.jsx]: Nice!!!!

File: superset/assets/javascripts/SqlLab/components/SqlEditor.jsx (hunk 1 above)
Comment (hughhhh) [reply superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: Ahhh true, the `func` param, i just don't like the bulky render view, but if there isn't a way to clean it up we can just go with it.

Comment (mistercrunch) [reply superset/assets/javascripts/SqlLab/components/SqlEditor.jsx]: moved to `this.getHotkeyConfig()`

**Comment by hughhhh** [comment]
🚢

**Pull request merged**